core/src/main/java/org/apache/carbondata/core/view/MVProvider.java

```diff
@@ -223,7 +224,7 @@ public void updateStatus(MVManager viewManager, List<MVSchema> schemaList,
 
   private void updateStatus(MVManager viewManager, String databaseName, List<MVSchema> schemaList,
       MVStatus status) throws IOException {
-    ICarbonLock carbonTableStatusLock = getStatusLock(databaseName);
+    ICarbonLock carbonTableStatusLock = getStatusLock(viewManager, databaseName);
```

Comment:
Can get databaseLocation and send only `databaseLocation` as parameter

Reply:
done